### ISSUE TITLE
How to keep original html or adjust SunEditor's standardized html function?

### ISSUE THREAD

**Issue by conquerorvn** [JiHong88/suneditor#577]
Hi SunEditor,
When add html to the editor, SunEditor will standardize the html by adding some tags, removing the attributes of some tags, especially the `<img>` and `<table>`.
For my project, I need SunEditor to keep the original Html? or adjust SunEditor's standardized html function?
Could you give me some guide.

**Comment by dreamsight** [comment]
Yes, interested in this one too, it also does this with the <div> tags, although managed to force a class that i can alter through a CSS file.

**Comment by JiHong88** [comment]
Sorry, for the late reply.😢
The `__se__tag` class prevents subtag changes.
can you show me the you html string?

**Comment by cscxj** [comment]
Is there a solution to this problem?

**Comment by cscxj** [comment]
@JiHong88 
My html:  
```html
<table>
  <tr>
    <td>
      <a href="#">link</a>
    </td>
  </tr>
</table>
```
After standardization:  
```html
<table>
  <tbody>
    <tr>
      <td>
        <div><a href="#">link</a></div>
      </td>
    </tr>
  </tbody>
</table>
```

**Comment by JiHong88** [comment]
@cscxj 
Currently, That format is a forced format.
Can't chang.

**Comment by AbdulkarimOgaji** [comment]
@JiHong88
not sure if this is related but im trying to add attributes in code view but when i toggle back in it disappears

`<h1 id="my-anchor">Introduction<h1>`

**Comment by JiHong88** [comment]
@AbdulkarimOgaji 
You can use the `attributesWhitelist` option.
ex)
```
attributesWhitelist: {
'all': 'style|data-.+', // Apply to all tags
input': 'checked|name' // Apply to input tag
 '???': '*' // "*" === all attributes
}
```

**Comment by eklect** [comment]
I tried that snippet above as well, and it does work for the `style` attribute, but not  `class`. I assume that's because `class` is part of the native attributes and it conflicts if allowed in the whitelist so using this:

```
'all': 'class|style',
```
This cold wouldn't actually work for `class` just `style`.  Is that assumption correct?

**Comment by glrph** [comment]
It seems the "class" attribute always gets removed even with 'all': '*'
Is there any workaround for this?

**Comment by aburd** [comment]
https://github.com/JiHong88/suneditor/issues/1320

As of 2.46.3, you should be able to use strictMode: false option, no?
Maybe this issue can be closed?

**Comment by mondjunge** [comment]
Using strictMode: false is causing several other issues. 

For instance the cursor is jumping when navigation with Arrow keys and you get breaks after Links with Mouseclicks in the editorarea.
